### ISSUE TITLE
Route refresh slow on master

### ISSUE THREAD

**Issue by dhammika** [Exa-Networks/exabgp#709]
##### ISSUE TYPE

 - Bug Report

##### OS
Linux/Debian 8.9

##### VERSION

```
ExaBGP : 4.0.2-1c737d99
Python : 3.4.2 (default, Oct  8 2014, 10:45:20)  [GCC 4.9.1]
```

##### ENVIRONMENT 

```
```

##### CONFIGURATION
```
process morebacon {
	encoder json;
	run /usr/bin/morebacon;
}

neighbor 10.1.1.2 {
    description "10.1.1.2";
    router-id 10.1.1.1;
    local-address 10.1.1.1;
    peer-as 666;
    local-as 666;
    hold-time 720;
    group-updates;
    capability {
	    asn4 enable;
	    add-path disable;
	    route-refresh enable;
    }
    family {
	    ipv4 unicast;
    }
    api {
	    processes [morebacon];
	    neighbor-changes;
	    receive {
	        parsed;
	        update;
	    }
    }
}
```

##### SUMMARY

Route refresh takes considerably longer on master, as much as 10x compared to 3.4 branch.

##### STEPS TO REPRODUCE
Peer with a router as ibgp neighbor.

##### EXPECTED RESULTS

Exa reactor loop shouldn't block on polling forked processes with pending peer messages.

##### ACTUAL RESULTS

Reactor loop seems to be blocking 10ms (select sleep below) on each iteration with pending upstream messages. fd `#6` is forked process stdout pipe and fd `#4` is peer tcp connection.
```
write(5, "{ \"exabgp\": \"4.0.1\", \"time\": 150"..., 722) = 722
wait4(1956130, 0x7fff2ecdb6b4, WNOHANG, NULL) = 0
select(7, [6], [], [], {0, 0})          = 0 (Timeout)
select(7, [6], [], [], {0, 10000})      = 0 (Timeout)
select(5, [4], [], [], {0, 0})          = 1 (in [4], left {0, 0})
recvfrom(4, "\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\x00\x66\x02", 19, 0, NULL, NULL) = 19
select(5, [4], [], [], {0, 0})          = 1 (in [4], left {0, 0})
recvfrom(4, "\x00\x00\x00\x4b\x40\x01\x01\x00\x40\x02\x12\x02\x04\x00\x00\x32\xbd\x00\x00\x23\x2a\x00\x00\xe0\x2d\x00\x03\x19\x81\x40\x03\x04"..., 83, 0, NULL, NULL) = 83
uname({sys="Linux", node="cheesecake", ...}) = 0
getppid()                               = 606758
write(5, "{ \"exabgp\": \"4.0.1\", \"time\": 150"..., 698) = 698
wait4(1956130, 0x7fff2ecdb6b4, WNOHANG, NULL) = 0
select(7, [6], [], [], {0, 0})          = 0 (Timeout)
select(7, [6], [], [], {0, 10000})      = 0 (Timeout)
select(5, [4], [], [], {0, 0})          = 1 (in [4], left {0, 0})
recvfrom(4, "\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\x00\x76\x02", 19, 0, NULL, NULL) = 19
select(5, [4], [], [], {0, 0})          = 1 (in [4], left {0, 0})
recvfrom(4, "\x00\x00\x00\x53\x40\x01\x01\x00\x40\x02\x16\x02\x05\x00\x00\x32\xbd\x00\x00\x1b\x1b\x00\x00\xd6\x8c\x00\x02\x04\x6c\x00\x02\x04"..., 99, 0, NULL, NULL) = 99
uname({sys="Linux", node="cheesecake", ...}) = 0
getppid()                               = 606758
write(5, "{ \"exabgp\": \"4.0.1\", \"time\": 150"..., 767) = 767
wait4(1956130, 0x7fff2ecdb6b4, WNOHANG, NULL) = 0
select(7, [6], [], [], {0, 0})          = 0 (Timeout)
select(7, [6], [], [], {0, 10000})      = 0 (Timeout)
select(5, [4], [], [], {0, 0})          = 1 (in [4], left {0, 0})
recvfrom(4, "\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\x02\xcd\x02", 19, 0, NULL, NULL) = 19
select(5, [4], [], [], {0, 0})          = 1 (in [4], left {0, 0})
recvfrom(4, "\x00\x00\x00\x4f\x40\x01\x01\x02\x40\x02\x0e\x02\x03\x00\x00\x32\xbd\x00\x00\x1b\x6a\x00\x00\x4e\x59\x40\x03\x04\xd1\xc5\x12\x01"..., 698, 0, NULL, NULL) = 698
```

##### IMPORTANCE

This is slowing down route refresh noticeably and stalling routing data pipeline.

**Comment by thomas-mangin** [comment]
Hopefully fixed with 24fca1cb (slight variant of your own fix)- if so please free to close.

**Comment by dhammika** [comment]
Nice, thanks.
